server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -491,6 +494,16 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                         // completion listener in this method. For the snapshot completion to work properly, the snapshot
                         // should still exist when listener is registered.
                         userCreateSnapshotListener.onResponse(snapshot.snapshot());
+
+                        if (hadAbortedInitializations) {
+                            final SnapshotsInProgress snapshotsInProgress = newState.custom(SnapshotsInProgress.TYPE);
+                            if (snapshotsInProgress != null) {
+                                final SnapshotsInProgress.Entry entry = snapshotsInProgress.snapshot(snapshot.snapshot());
+                                if (entry != null) {
```

Comment:
we should be able to assert != null here?

Reply:
I guess I was paranoid about potentially throwing an NPE in the cluster state thread :D but yea, `assert` should in theory be just fine => moving to `assert` :)